### PR TITLE
Handler class base framework

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
The class is responsible for handling VPD data w.r.t a given FRU.
Major responsibility of the class:
- Call factory to pick appropriate parser.
- Provide functionality to get parsed VPD map.
- Process VPD data to arrange under interfaces.
- Publish VPD on DBus.

### CODE REVIEW

File: include/handler.hpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    /**
+     * @brief An API to process VPD.
+     *
+     * This api is exposed to process VPD map recieved as a parameter and call
+     * respective API to populate VPD under interfaces and publish them on
+     * D-Bus.
+     *
+     * @param[in] vpdFilePath - EEPROM path of the FRU whose data needs to be
+     * handled.
+     * @param[in] vpdMap - Map of parsed VPD data.
```
Comment (nkantesh) [inline include/handler.hpp:53]: Doxygen is missing for ParsedVPD parameter, please add it.

File: include/handler.hpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+     *
+     * The api can be called to get map of parsed VPD data in the format as
+     * defined by the type returned.
+     *
+     * Note: Empty map is returned in case of any error. Caller responsibility
+     * to handle accordingly.
+     *
+     * @param[in] vpdFilePath - EEPROM path of FRU whose data needs to be
+     * parsed.
+     * @param[out] vpdMap - Parsed VPD map.
```
Comment (nkantesh) [inline include/handler.hpp:70]: Doxygen is missing for ParsedVPD parameter, please add it.

File: include/handler.hpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    void
+        getVPDMap(std::string_view vpdFilePath,
+                  std::variant<types::ParsedVPD, types::KeywordVpdMap>& vpdMap);
+
+  private:
+    /**
+     * @brief API to get VPD data in a uint8_t vector.
```
Comment (nkantesh) [inline include/handler.hpp:78]: please remove uint8_t .. just vector should be fine

File: include/types.hpp [hunk 4]
```diff
@@ -36,5 +36,11 @@ using DbusVariantType = std::variant<
  using MapperGetObject =
     std::vector<std::pair<std::string, std::vector<std::string>>>;
 
+using ParsedVPD = std::unordered_map<std::string,
+                                  std::unordered_map<std::string, std::string>>;
+using BinaryArray = std::vector<uint8_t>;
```
Comment (nkantesh) [inline include/types.hpp:41]: Better to use the name BinaryVector for better readability  instead of BinaryArray

File: include/handler.hpp (hunk 1 above)
Comment (SunnySrivastava1984) [reply include/handler.hpp:53]: ParsedVPD is not a parameter. Doxygen is not required for that.

File: include/handler.hpp (hunk 3 above)
Comment (SunnySrivastava1984) [reply include/handler.hpp:78]: Vector can be of anything. Having the type gives better clarity.

File: include/handler.hpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+     *
+     * Some FRU has entry for offset from where VPD data starts. This api reads
+     * the VPD data into a uint8_t vector, corresponding to offset w.r.t that
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/handler.hpp:81]: Vector can be of anything. Having the type, gives better clarity.

File: include/handler.hpp [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+     * FRU.
+     *
+     * Note: Caller responsibility to check for validity of the VPD vector.
+     *
+     * @param[in] vpdFilePath - EEPROM path of FRU.
+     * @param[out] vpdVector - VPD data, read into an uint8_t vector.
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/handler.hpp:87]: Explaining the type makes it easy for the reader to understand what BInaryArray means as it is not a standard type.

**Review by jinuthomas — CHANGES_REQUESTED** [review]
Has conflict , could you resolve it for it to be reviewed

**Comment by SunnySrivastava1984** [comment]
Not required. Code merged via different commit.

**Pull request closed without merging**